sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -459,6 +459,7 @@ object LimitPushDown extends Rule[LogicalPlan] {
       val newJoin = joinType match {
         case RightOuter => join.copy(right = maybePushLocalLimit(exp, right))
         case LeftOuter => join.copy(left = maybePushLocalLimit(exp, left))
+        case Cross => join.copy(left = maybePushLocalLimit(exp, left), right = maybePushLocalLimit(exp, right))
```

Comment:
how about inner join without condition?

Reply:
maybe we  can match `InnerLike` when condition is empty.

Reply:
A = {(a, 0), (b, 1), (c, 2), (d, 0), **(e, 1), (f, 2)**}
B = {**(e, 1), (f, 2)**}

A inner join B limit 2
If there is limit 2, (a, 0), (b, 1) inner join {(e, 1), (f, 2)}, the result is empty. But the real result is not empty.

Reply:
When set  spark.sql.crossJoin.enabled=true, 
inner join without condition, LeftOuter without condition, RightOuter without condition, FullOuter without condition， all these are  iterally cross join?

Reply:
+1

Reply:
@guoxiaolongzte nope.

Reply:
@cloud-fan 
Please give me some advice. Thank you.

Reply:
if there is no join condition, I think join type doesn't matter and we can always push down limits.

Reply:
There are two tables as follows：
CREATE TABLE `**test1**`(`id` int, `name` int);
CREATE TABLE `**test2**`(`id` int, `name` int);

test1 table data:
2,2
1,1

test2 table data:
2,2
3,3
4,4

Execute sql select * from test1 t1 **left anti join** test2 t2 on t1.id=t2.id limit 1; The result:
1,1

But 
   we push the limit 1 on left side, the result is not correct. Result is empty.
   we push the limit 1 on right side, the result is not correct. Result is empty.

So
    left anti join no need to push down limit. Similarly, left semi join is the same logic.